src/disc/stk/Albany_STKDiscretization.cpp

```diff
@@ -2024,12 +2029,71 @@ STKDiscretization::computeSideSets()
       lssList[ss_key].numSides       = globalSideSetViews[ss_key].max_sides;
       lssList[ss_key].numCellsOnSide = Kokkos::subview(globalSideSetViews[ss_key].numCellsOnSide, current_index, Kokkos::ALL() );
       lssList[ss_key].cellsOnSide    = Kokkos::subview(globalSideSetViews[ss_key].cellsOnSide,    current_index, Kokkos::ALL(), Kokkos::ALL() );
+      lssList[ss_key].sideSetIdxOnSide    = Kokkos::subview(globalSideSetViews[ss_key].sideSetIdxOnSide,    current_index, Kokkos::ALL(), Kokkos::ALL() );
 
       current_local_index[ss_key]++;
 
       ss_it++;
     }
   }
+
+  // 6) Populate localDOFViews for GatherVerticallyContractedSolution
+  const int maxSideNodes = 6;
```

Comment:
Do you set '6' cause you're thinking about quadratic triangles? I don't think we support quadratic elements in Albany. But then, if we did, 6 would not fit quadratic Quads, which would show up when using prisms.

You could be more tight, and loop over the cell_topo sides, and count their nodes. That will tell you exactly what 'max_side_nodes' should be. I suspect 6 is an overshoot for most of our cases. A tighter upper bound might help with caching.

Reply:
Yeah, this was a bandaid thing that I didn't get around to fixing. I'll loop over the sides to get the tightest bound here.